lib/cl/memory/bda.jl

```diff
@@ -5,10 +5,12 @@ struct BufferDeviceMemory <: AbstractMemory
     context::Context
 end
 
+BufferDeviceMemory() = BufferDeviceMemory(C_NULL, CL_NULL, 0, context())
+
 function bda_alloc(bytesize::Integer;
         alignment::Integer = 0, device_access::Symbol = :rw, host_access::Symbol = :rw
     )
-    bytesize == 0 && error("size 0 is not supported for BufferDeviceMemory.")
```

Comment:
Can you do this from `alloc` in `src/memory.jl` instead? We probably ought to do this for other memories as well, in order to flush out issues more quickly, and be compatible with hypothetical drivers not supporting zero-size allocations (unless this is mandated by the spec).